src/mono/wasm/build/WasmApp.targets

```diff
@@ -172,7 +173,7 @@
     <!-- Use a unique property, so the already run wasm targets can also run -->
     <MSBuild Projects="$(MSBuildProjectFile)"
```

Comment:
I'm curious how this works.
Do we run separate MSBuild process here (which doesn't share variables and items from parent ?)
Do we still need to have it as separate process ?
Do we still need `_WasmInNestedPublish_UniqueProperty_XYZ` ?

Reply:
> Do we run separate MSBuild process here (which doesn't share variables and items from parent ?)

No, but it runs it in a fresh context. I will be removing this, but I'm waiting for the blazor<->runtime build changes to happen.

> Do we still need to have it as separate process ?

It's not necessarily a separate process. And this PR doesn't change anything about this.

> Do we still need _WasmInNestedPublish_UniqueProperty_XYZ ?

Yep

Essentially, this PR doesn't change any behavior w.r.t the nested publish. The nested publish bit was a hacky solution, and will be removed. But I'm holding that off till we get further on the build consolidation with blazor.

Reply:
@ilonatommy opened issue for ICU related dependency on not nesting it 
https://github.com/dotnet/runtime/issues/82819